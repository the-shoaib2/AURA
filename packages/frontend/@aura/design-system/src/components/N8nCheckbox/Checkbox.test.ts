import { render } from '@testing-library/vue';

import auraCheckbox from './Checkbox.vue';

describe('components', () => {
	describe('auraCheckbox', () => {
		it('should render without label and child content', () => {
			const { container } = render(auraCheckbox);
			expect(container).toMatchSnapshot();
		});

		it('should render with label', () => {
			const { container } = render(auraCheckbox, { props: { label: 'Checkbox' } });
			expect(container).toMatchSnapshot();
		});

		it('should render with child', () => {
			const { container } = render(auraCheckbox, {
				slots: { default: '<strong>Bold text</strong>' },
			});
			expect(container).toMatchSnapshot();
		});

		it('should render with both child and label', () => {
			const { container } = render(auraCheckbox, {
				props: { label: 'Checkbox' },
				slots: { default: '<strong>Bold text</strong>' },
			});
			expect(container).toMatchSnapshot();
		});
	});
});
