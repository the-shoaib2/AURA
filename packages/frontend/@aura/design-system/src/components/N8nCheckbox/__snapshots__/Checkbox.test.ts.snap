// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > auraCheckbox > should render with both child and label 1`] = `
<div>
  <label
    class="el-checkbox aura-checkbox auraCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <input
        class="el-checkbox__original"
        type="checkbox"
        value="Checkbox"
      />
      <span
        class="el-checkbox__inner"
      />
    </span>
    <span
      class="el-checkbox__label"
    >


      <strong>
        Bold text
      </strong>

      <div
        class="container"
        data-test-id="input-label"
      >
        <div
          class="labelRow"
        >
          <label
            class="aura-input-label inputLabel heading medium"
          >
            <div
              class="main-content"
            >
              <div
                class="title"
              >
                <span
                  class="aura-text size-medium regular"
                >

                  Checkbox
                  <!--v-if-->

                </span>
              </div>
              <!--v-if-->
            </div>
            <div
              class="trailing-content"
            >
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </div>
          </label>
          <!--v-if-->
        </div>


      </div>

      <!--v-if-->
    </span>
  </label>
</div>
`;

exports[`components > auraCheckbox > should render with child 1`] = `
<div>
  <label
    class="el-checkbox aura-checkbox auraCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <input
        class="el-checkbox__original"
        type="checkbox"
      />
      <span
        class="el-checkbox__inner"
      />
    </span>
    <span
      class="el-checkbox__label"
    >


      <strong>
        Bold text
      </strong>

      <!--v-if-->

      <!--v-if-->
    </span>
  </label>
</div>
`;

exports[`components > auraCheckbox > should render with label 1`] = `
<div>
  <label
    class="el-checkbox aura-checkbox auraCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <input
        class="el-checkbox__original"
        type="checkbox"
        value="Checkbox"
      />
      <span
        class="el-checkbox__inner"
      />
    </span>
    <span
      class="el-checkbox__label"
    >



      <div
        class="container"
        data-test-id="input-label"
      >
        <div
          class="labelRow"
        >
          <label
            class="aura-input-label inputLabel heading medium"
          >
            <div
              class="main-content"
            >
              <div
                class="title"
              >
                <span
                  class="aura-text size-medium regular"
                >

                  Checkbox
                  <!--v-if-->

                </span>
              </div>
              <!--v-if-->
            </div>
            <div
              class="trailing-content"
            >
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </div>
          </label>
          <!--v-if-->
        </div>


      </div>

      <!--v-if-->
    </span>
  </label>
</div>
`;

exports[`components > auraCheckbox > should render without label and child content 1`] = `
<div>
  <label
    class="el-checkbox aura-checkbox auraCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <input
        class="el-checkbox__original"
        type="checkbox"
      />
      <span
        class="el-checkbox__inner"
      />
    </span>
    <span
      class="el-checkbox__label"
    >


      <!--v-if-->
    </span>
  </label>
</div>
`;
